src/boot/efi/stub.c

```diff
@@ -26,19 +26,30 @@ DECLARE_NOALLOC_SECTION(".sdmagic", "#### LoaderInfo: systemd-stub " GIT_VERSION
 
 DECLARE_SBAT(SBAT_STUB_SECTION_TEXT);
 
-static EFI_STATUS combine_initrd(
-                EFI_PHYSICAL_ADDRESS initrd_base, size_t initrd_size,
+/* Combine initrds by concatenation in memory.
+ * uki_initrds come from sections in the UKI and will be padded,
+ * extra initrds are generated and don't need further padding */
+static EFI_STATUS combine_initrds(
+                EFI_PHYSICAL_ADDRESS uki_initrds[], const size_t uki_initrd_sizes[], size_t n_uki_initrds,
                 const void * const extra_initrds[], const size_t extra_initrd_sizes[], size_t n_extra_initrds,
```

Comment:
please just make this one set of arrays. and apply the padding always. if the initrds happen to be padded already it's a NOP to pad them again after all.